### PR TITLE
feat(extension): add embed video to mnemonic intro step

### PR DESCRIPTION
**Pull request by jplorek-atix**
# Checklist

- [ ] JIRA - \<[LW-8699](https://input-output.atlassian.net/browse/LW-8699)>
- [ ] Proper tests implemented
- [ ] Screenshots added.

---

## Proposed solution

This PR implements an embed video to the mnemonic intro step in create wallet flow
also, adds a post hog event `onboarding | new wallet | passphrase intro | play video | click`, this event is sent when click over the video to play it the first time

## Screenshots

Screenshot 2023-10-12 at 10 12 59

[LW-8699]: https://input-output.atlassian.net/browse/LW-8699?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

---
# Allure report
`allure-report-publisher` generated test report!

**smokeTests**: ✅ [test report](https://lace-qa:8PP5wBaDV6UoXj@dq4ajm5i5q7bz.cloudfront.net/linux/chrome/2527/6499730164/index.html) for [11c9e355](https://github.com/input-output-hk/lace/pull/643/commits/11c9e355fd1c8d9390143fcd11e86b33c48c78eb)
|       | passed | failed | skipped | flaky | total | result |
|-------|--------|--------|---------|-------|-------|--------|
| Total | 32     | 0      | 0       | 0     | 32    | ✅     |

### CODE REVIEW

**Review by DominikGuzei — APPROVED** [review]
Code LGTM 🚢

File: packages/core/src/ui/components/WalletSetup/WalletSetupMnemonicIntroStep.tsx [hunk 1]
```diff
@@ -1,28 +1,50 @@
...
+          setShouldDisplayThumbnail(false);
+          videoRef.current.src += '&autoplay=1';
+          onClickVideo();
+        }}
+      >
+        {shouldDisplayThumbnail && <div className={styles.overlay} />}
+        <iframe
+          ref={videoRef}
+          className={styles.video}
+          src="https://www.youtube-nocookie.com/embed/hOFVXo969rk?si=0a-hNDVME6eTboIX"
```
Comment (marcin-mazurek) [inline packages/core/src/ui/components/WalletSetup/WalletSetupMnemonicIntroStep.tsx:43]: Do you think it makes sense to follow the pattern of configuring URLs through environment variables? https://github.com/input-output-hk/lace/blob/main/apps/browser-extension-wallet/.env.defaults#L30

File: packages/core/src/ui/components/WalletSetup/WalletSetupMnemonicIntroStep.tsx [hunk 2]
```diff
@@ -1,28 +1,50 @@
...
+      description={translations.description}
+      linkText={translations.linkText}
+      onBack={onBack}
+      onNext={onNext}
+      currentTimelineStep={WalletTimelineSteps.RECOVERY_PHRASE}
+    >
+      <div
+        className={styles.videoContainer}
+        onClick={() => {
+          setShouldDisplayThumbnail(false);
```
Comment (marcin-mazurek) [inline packages/core/src/ui/components/WalletSetup/WalletSetupMnemonicIntroStep.tsx:34]: Do we still need this? Is that required to send an event to PostHog? Can we instead have `onclick` event handler on the iframe?

File: packages/core/src/ui/components/WalletSetup/WalletSetupOption.module.scss [hunk 3]
```diff
@@ -46,3 +46,38 @@
...
+.playIcon {
+  position: absolute;
+  top: 50%;
+  left: 50%;
+  transform: translate(-50%, -50%);
+  width: 97px;
+  height: 97px;
+  z-index: 2;
+}
+
```
[the remark this answers is not in the document]
Comment (jplorek-atix) [reply packages/core/src/ui/components/WalletSetup/WalletSetupOption.module.scss]: done in [93b4477](https://github.com/input-output-hk/lace/pull/643/commits/93b447741b535caca99e0584f580cd7323043e28)

File: packages/core/src/ui/components/WalletSetup/WalletSetupMnemonicIntroStep.tsx (hunk 1 above)
Comment (jplorek-atix) [reply packages/core/src/ui/components/WalletSetup/WalletSetupMnemonicIntroStep.tsx]: done in [93b4477](https://github.com/input-output-hk/lace/pull/643/commits/93b447741b535caca99e0584f580cd7323043e28)

File: packages/core/src/ui/components/WalletSetup/WalletSetupMnemonicIntroStep.tsx (hunk 2 above)
Comment (jplorek-atix) [reply packages/core/src/ui/components/WalletSetup/WalletSetupMnemonicIntroStep.tsx:36]: yes, is required to send the event to PostHog, unfortunately the iframe has a separate document tree, so the events that bubble through its tree finish at the root of the iframe document, so we cannot listen for those events in the outer context of the iframe

Comment (marcin-mazurek) [reply packages/core/src/ui/components/WalletSetup/WalletSetupMnemonicIntroStep.tsx:36]: Ok - then just one more suggestion to call it `setShouldDisplayOverlay` or maybe even shorter, `setOverlayVisible` (`should` somehow reminds me of `shouldComponentUpdate` 😁 )

Comment (jplorek-atix) [reply packages/core/src/ui/components/WalletSetup/WalletSetupMnemonicIntroStep.tsx]: done in [11c9e35](https://github.com/input-output-hk/lace/pull/643/commits/11c9e355fd1c8d9390143fcd11e86b33c48c78eb)

**Review by marcin-mazurek — APPROVED** [review]

**Review by bslabiak — APPROVED** [review]

**Pull request merged**